### PR TITLE
avm2: Added flash.net.URLVariables

### PR DESCRIPTION
**Pull request by dowgird**
At this point it handles encoding only.

### CODE REVIEW

File: tests/tests/swfs/avm2/url_vars/Test.as~ [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package {
```
Comment (relrelb) [inline tests/tests/swfs/avm2/url_vars/Test.as~:1]: I guess this is a leftover file...

File: core/src/avm2/globals/flash/net/URLVariables.as [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package flash.net {
+    import flash.utils.escapeMultiByte;
+    public dynamic class URLVariables {
+	// TODO: construct from String
+        private var _url: String;
+	public function URLVariables(url:String = null) {
+	    this._url = url;
+	}
+	public function set url(value:String): void {
```
Comment (relrelb) [inline core/src/avm2/globals/flash/net/URLVariables.as:9]: I don't see any `url` property in the [docs](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/net/URLVariables.html). I do see however `decode()`, which is missing in this PR.

File: tests/tests/swfs/avm2/url_vars/Test.as~ (hunk 1 above)
Comment (dowgird) [reply tests/tests/swfs/avm2/url_vars/Test.as~]: Removed!

File: core/src/avm2/globals/flash/net/URLVariables.as (hunk 2 above)
Comment (dowgird) [reply core/src/avm2/globals/flash/net/URLVariables.as]: I added a check and a TODO to make it clear what is not implemented yet (basically it currently works as an encoder but not as a decoder).

Comment (dowgird) [reply core/src/avm2/globals/flash/net/URLVariables.as]: Update: had to remove the check because the linter complained. Left TODOs.

**Review by Aaron1011 — APPROVED** [review]

**Pull request merged**